fedimint-api/src/tiered_multi.rs

```diff
@@ -277,7 +296,29 @@ where
 mod test {
     use fedimint_api::Amount;
 
-    use crate::TieredMulti;
+    use crate::{Tiered, TieredMulti};
+
+    #[test]
+    fn represent_amount_targets_denomination_sets() {
+        let starting = coins(vec![
```

Comment:
This existing `coins` function seems meh, and unintuitive, and I needed to read it to understand what this part does. No idea why we need `0_usize` there. Would be better as:

```
let starting_note_counts = Tiered::from_iter((Amount::from_sat(1), 1), ...);
```

That `coins` doesn't seem to save much space anyway.

We need to clean up this `Amount` stuff too. All these conversions everywhere are annoying. (See #general post on Discord).